File: backend/Node.JS/calindra-desafio/src/controllers/v1.0/geolocation/commands/handlers/calculator/calculator.handler.ts
import { Injectable } from '@nestjs/common';
import { Observable } from 'rxjs';

import { CommandHandler } from '../command-handler';
import { DistanceCalculatorService } from '../../../services/distance-calculator/distance-calculator.service';
import { GeolocationReferencesInfo } from '../../../models/responses/geolocation-references-info';
import { Addresses } from '../../inputs/addresses';
import { MethodResult } from 'src/shared/results/method-result';


@Injectable()
export class CalculatorHandler implements CommandHandler<Addresses> {

    constructor(private distanceCalculatorService: DistanceCalculatorService) { }

    execute(command: Addresses): Observable<MethodResult> {
        return this.distanceCalculatorService.calculateDistancesFromAddressess(command);
    }
}

File: backend/.NET/Calindra.Desafio/Calindra.Desafio.Domain/Commands/Handlers/CalculatorHandler.cs
using Calindra.Desafio.Domain.Commands.Inputs;
using Calindra.Desafio.Domain.Results;
using Calindra.Desafio.Domain.Services;
using System.Threading.Tasks;

namespace Calindra.Desafio.Domain.Commands.Handlers
{
    public class CalculatorHandler : ICommandHandler<Addresses>
    {
        private readonly DistanceCalculatorService _service;

        public CalculatorHandler(DistanceCalculatorService service)
        {
            _service = service;
        }

        public async Task<MethodResult> Execute(Addresses command)
        {
            return await _service.CalculateDistancesFromAddressess(command);
        }
    }
}
